sphinx/ext/autodoc/__init__.py

```diff
@@ -177,7 +177,7 @@ def merge_members_option(options: dict) -> None:
 
 # Some useful event listener factories for autodoc-process-docstring.
 
-def cut_lines(pre: int, post: int = 0, what: str | None = None) -> Callable:
+def cut_lines(pre: int, post: int = 0, what: str | list[str] | None = None) -> Callable:
```

Comment:
You should also update the signature of `between` below since they are hooks for the same event (but you can also do it in a follow-up PR since it's not directly tight to `docs/conf.py` itself).